client/state/data-layer/wpcom/sites/automated-transfer/eligibility/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	AUTOMATED_TRANSFER_ELIGIBILITY_REQUEST,
+} from 'state/action-types';
+
+import { updateEligibility } from 'state/automated-transfer/actions';
+import {
+	AUTOMATED_TRANSFER_STATUS,
+	eligibilityHolds,
+} from 'state/automated-transfer/constants';
+
+const eligibilityUrl = domain =>
+	`https://public-api.wordpress.com/rest/v1/sites/${ domain }/automated-transfers/eligibility`;
+
+const statusMapping = {
```

Comment:
We could avoid this mapping by defining `eligibilityHolds` like this in the first place:
```js
// state/automated-transfer/constants.js
const ELIGIBILITY_HOLDS = {
  no_wpcom_nameservers: 'NO_WPCOM_NAMESERVERS',
  ...
}
```

And then in `fromApi`:
```js
  eligibilityHolds: get( ELIGIBILITY_HOLDS, data.status_code, [] ), 
```

(Also, by uppercasing the name, it would still convey the "true" constant meaning and avoid confusion between the constant and the `fromApi` return, but that's not really relevant here.)